pkg/watcher/conditionwatcher.go

```diff
@@ -116,11 +116,12 @@ func (c *ConditionWatcher) NewConditionWaiter(ctx context.Context, objType Objec
 		waitChan:      make(chan waitResult, 1),
 	}
 
-	_, exists := c.waitMap.LoadOrStore(getTypedName(objType, objName), &entry)
+	key := getTypedName(objType, objName)
+	val, exists := c.waitMap.LoadOrStore(key, map[*waitEntry]struct{}{&entry: {}})
```

Comment:
I believe you'll need to use a `sync.Map` instead of a normal map. The unit tests are failing due to data races with this map.

Reply:
The stored value isn't - that's what also need to be a `sync.Map`. Sorry, I wasn't clear.